server/base/src/main/java/org/apache/accumulo/server/util/ECAdmin.java

```diff
@@ -130,6 +130,7 @@ public void execute(final String[] args) {
 
   private void cancelCompaction(ServerContext context, String ecid) {
     CompactionCoordinatorService.Client coordinatorClient = null;
+    ecid = ExternalCompactionUtil.prefixECID(ecid);
```

Comment:
I assume you are trying to clean up bad user input here? Can we just use `ExternalCompactionId.of(String)` here and let the user pass the right thing? Maybe we can clean up the error messages in that method to be more specific about the issues.

Reply:
> I assume you are trying to clean up bad user input here?

Yes. 

> Can we just use ExternalCompactionId.of(String) here and let the user pass the right thing? Maybe we can clean up the error messages in that method to be more specific about the issues.

I wasn't sure. I did this in the monitor just because it wasn't clear to the user that the server code required the prefix and I didn't like relying on trial and error to figure that out. Even with a nice error message, I don't like users having to guess if you need the prefix or not.

I pinged @keith-turner on chat asking if he had any thoughts behind the origin of the prefix but haven't heard back yet.

Reply:
`ExternalCompactionId.of` also validates that the id is a UUID. If we are going to have another method to take a String and massage it into an ExternalCompactionId, then maybe we can move your method body into `ExternalCompactionId.from(String)` or something like that. Then at least we can re-use PREFIX.

Reply:
I like the idea of consolidating the logic into `ExternalCompactionId`. The only issue is that both places I need a String and not the object. What about making the prefix method static member of `ExternalCompactionId`? Then we could reuse the validation in the `of()` method.

Reply:
Right, but on the server side, it parses the incoming ecid string into an ExternalCompactionId. It would be good to know that on the client side it's a valid ExternalCompactionId before sending it. You can always get the String value from the ExternalCompactionId, right? To be clear, I'm suggesting something like `ExternalCompactionId.from(String).toString()`. Whether you call the method `from`,`parse`, etc. doesn't matter to me. You called the method `prefix`, but it looks like it's just trying to create a valid ECID from the user input.

Reply:
Created `from` method in b233afe